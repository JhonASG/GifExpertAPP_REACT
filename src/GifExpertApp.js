import React, { useState } from 'react';
import {AddCategory, GifGrid} from './components';

const GifExpertApp = ({ defaultCategory = [] }) => {
    const title = 'GifExpertApp';
    // const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState( defaultCategory );
    
    /*const handleAddCategory = () => {
        // Forma de agregar las categorias con el operador spreed ...Name
        // const newElement = 'Naruto';
        // setCategories([...categories, newElement]);
        const newElement = ['Naruto'];
        setCategories(categories.concat(newElement));
    }*/

    return (
        <>
            <h2>{ title }</h2>
            <AddCategory setCategories = {setCategories} categories = {categories}/>
            <hr />
            <ol>
                {
                    // El map() para transformar cada uno de los elementos del arreglo
                    // map() recibe dos argumentos el valor del elemento del arreglo y el indice de ese elemento
                    // el key sirve para que React sepa que elemento esta renderizando
                    categories.map((category) => {
                        return (
                            /* Nos traemos el componente GifGrid que va ser el encargado de recibir la entrada
                            de texto mediante el input text y realizar una petici√≥n http a una API. */
                            <GifGrid key={category} category = {category}/>
                        )
                    })
                }
            </ol>
            {/*<button onClick={handleAddCategory}>Agregar Serie</button>*/}
        </>
    )
}

export default GifExpertApp;