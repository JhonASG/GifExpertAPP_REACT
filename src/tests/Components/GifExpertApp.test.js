import React from "react";
import GifExpertApp from "../../GifExpertApp";
import { render, screen } from "@testing-library/react";

describe('Prueba en el componente GifExpertApp', () => {
    test('Debe renderizar correctamente el componente', () => {
        const view = render( <GifExpertApp /> );
        
        expect(view).toMatchSnapshot();
    });

    test('Debe mostrar una lista de categorias', () => {
        const categories = ["Naruto", "Dragon Ball", "One Punch", "Marvel"];

        const view = render( <GifExpertApp defaultCategory={ categories } /> );

		expect(screen.queryByRole('list').querySelectorAll('h3').length).toBe(categories.length);
        expect(view).toMatchSnapshot();
    })
});