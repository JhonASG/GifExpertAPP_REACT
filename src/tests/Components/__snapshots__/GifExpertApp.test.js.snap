// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en el componente GifExpertApp Debe mostrar una lista de categorias 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        GifExpertApp
      </h2>
      <form
        data-testid="form"
      >
        <input
          type="text"
          value=""
        />
      </form>
      <hr />
      <ol>
        <h3
          class="animate__animated animate__bounce"
        >
          Naruto
        </h3>
        <p
          class="animate__animated animate__flash"
          data-testid="carga"
        >
          Loading ...
        </p>
        <div
          class="card-grid"
        />
        <h3
          class="animate__animated animate__bounce"
        >
          Dragon Ball
        </h3>
        <p
          class="animate__animated animate__flash"
          data-testid="carga"
        >
          Loading ...
        </p>
        <div
          class="card-grid"
        />
        <h3
          class="animate__animated animate__bounce"
        >
          One Punch
        </h3>
        <p
          class="animate__animated animate__flash"
          data-testid="carga"
        >
          Loading ...
        </p>
        <div
          class="card-grid"
        />
        <h3
          class="animate__animated animate__bounce"
        >
          Marvel
        </h3>
        <p
          class="animate__animated animate__flash"
          data-testid="carga"
        >
          Loading ...
        </p>
        <div
          class="card-grid"
        />
      </ol>
    </div>
  </body>,
  "container": <div>
    <h2>
      GifExpertApp
    </h2>
    <form
      data-testid="form"
    >
      <input
        type="text"
        value=""
      />
    </form>
    <hr />
    <ol>
      <h3
        class="animate__animated animate__bounce"
      >
        Naruto
      </h3>
      <p
        class="animate__animated animate__flash"
        data-testid="carga"
      >
        Loading ...
      </p>
      <div
        class="card-grid"
      />
      <h3
        class="animate__animated animate__bounce"
      >
        Dragon Ball
      </h3>
      <p
        class="animate__animated animate__flash"
        data-testid="carga"
      >
        Loading ...
      </p>
      <div
        class="card-grid"
      />
      <h3
        class="animate__animated animate__bounce"
      >
        One Punch
      </h3>
      <p
        class="animate__animated animate__flash"
        data-testid="carga"
      >
        Loading ...
      </p>
      <div
        class="card-grid"
      />
      <h3
        class="animate__animated animate__bounce"
      >
        Marvel
      </h3>
      <p
        class="animate__animated animate__flash"
        data-testid="carga"
      >
        Loading ...
      </p>
      <div
        class="card-grid"
      />
    </ol>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Prueba en el componente GifExpertApp Debe renderizar correctamente el componente 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2>
        GifExpertApp
      </h2>
      <form
        data-testid="form"
      >
        <input
          type="text"
          value=""
        />
      </form>
      <hr />
      <ol />
    </div>
  </body>,
  "container": <div>
    <h2>
      GifExpertApp
    </h2>
    <form
      data-testid="form"
    >
      <input
        type="text"
        value=""
      />
    </form>
    <hr />
    <ol />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
